### PR TITLE
Add release notes for `1.0.7`

### PR DESCRIPTION
**Pull request by fluiddot**
Related to pfHvTO-aQ-p2.

## Proposed Changes

- Add release notes for changes introduced in version `1.0.7` that aren't behind feature flags (https://github.com/Automattic/studio/compare/v1.0.6...trunk).

## Testing Instructions

N/A

## Pre-merge Checklist

- [x] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -1,3 +1,10 @@
+1.0.7
+=====
+* Update instructions of translation process #381
```
Comment (wojtekn) [inline RELEASE-NOTES.txt]: Let's remove this one, as the change was made in the readme and not in the app.

File: RELEASE-NOTES.txt [hunk 2]
```diff
@@ -1,3 +1,10 @@
...
+* Remove test timers related to code block #380
```
Comment (wojtekn) [inline RELEASE-NOTES.txt]:
```suggestion
* Improve test timers handling #380
```

File: RELEASE-NOTES.txt [hunk 3]
```diff
@@ -1,3 +1,10 @@
...
+* Avoid data corruption due to writing concurrently in the user data file #366
+* Allow blocking user from adding demo sites #354
```
Comment (wojtekn) [inline RELEASE-NOTES.txt]:
```suggestion
* Display a meaningful message for users who has demo sites blocked #354
```

Comment (katinthehatsite) [reply RELEASE-NOTES.txt]: "for users who have" ( not has)

Perhaps we could also use something like:

"Display a message for users who don't have demo sites available for their account"

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (fluiddot) [reply RELEASE-NOTES.txt]: This has been addressed in https://github.com/Automattic/studio/pull/412/commits/be5c825387362755911d8fb8897b32a8decc8165.

File: RELEASE-NOTES.txt (hunk 2 above)
Comment (fluiddot) [reply RELEASE-NOTES.txt]: This has been addressed in https://github.com/Automattic/studio/pull/412/commits/be5c825387362755911d8fb8897b32a8decc8165.

File: RELEASE-NOTES.txt (hunk 3 above)
Comment (fluiddot) [reply RELEASE-NOTES.txt]: Thanks for the suggestions 🙇 ! I've updated this entry in https://github.com/Automattic/studio/pull/412/commits/be5c825387362755911d8fb8897b32a8decc8165.

**Review by wojtekn — APPROVED** [review]

**Pull request merged**